readthedocs/doc_builder/config.py

```diff
@@ -0,0 +1,114 @@
+
+from readthedocs_build.config import (ConfigError, BuildConfig,
+                                      load as load_config)
+
+
+class ConfigWrapper(object):
+
+    """
+    A config object that wraps the Project & YAML based configs.
+
+    Gives precidence to YAML, falling back to project if it isn't defined.
+
+    We only currently implement a subset of the existing YAML config.
+    This should be the canonical source for our usage of the YAML files,
+    never accessing the config object directly.
+
+    """
+
+    def __init__(self, version, yaml_config):
+        self._version = version
+        self._project = version.project
+        self._yaml_config = yaml_config
+
+    @property
+    def install_project(self):
+        if 'setup_py_install' in self._yaml_config.get('python', {}):
+            return self._yaml_config['python']['setup_py_install']
+        else:
+            return self._project.install_project
+
+    @property
+    def python_interpreter(self):
+        if 'version' in self._yaml_config.get('python', {}):
+            ver = self._yaml_config['python']['version']
+            if str(ver).startswith('2'):
+                return 'python'
+            else:
+                return 'python3'
+        else:
+            return self._project.python_interpreter
+
+    @property
+    def python_version(self):
+        if 'version' in self._yaml_config.get('python', {}):
+            ver = self._yaml_config['python']['version']
+            return ver
+        else:
+            if self._project.python_interpreter == 'python':
```

Comment:
This is minor, but this code might make more sense with `python_interpreter` relying on the result of `python_version`, rather than the inverse. `python_version` is returning an integer interpretation anyways, so would reduce some of the roundabout logic here.